src/hooks/useKeyCodeActions.js

```diff
@@ -0,0 +1,22 @@
+import { useCallback, useMemo, useEffect, useState } from 'react'
+import { saveAs } from 'file-saver'
+import { generateRandomNumber } from '../lib/math-utils'
+import { dateFormat } from '../utils/date-utils'
+
+export default function useKeyCodeActions() {
```

Comment:
As I was suggesting in another comment, I think we should rename this to be `useOneTimeCodeActions()` (or the name we decide).